src/NHibernate.Test/DriverTest/FirebirdClientDriverFixture.cs

```diff
@@ -14,157 +14,160 @@ public class FirebirdClientDriverFixture
 
-			//open the second connection
-			connection2.Open();
-			VerifyCountOfEstablishedConnectionsIs(allreadyEstablished + 1, "After second open");
+				//open the second connection
+				connection2.Open();
+				VerifyCountOfEstablishedConnectionsIs(allreadyEstablished + 1, "After second open");
 
-			//return it to the pool
-			connection2.Close();
-			VerifyCountOfEstablishedConnectionsIs(allreadyEstablished + 1, "After second close");
+				//return it to the pool
+				connection2.Close();
+				VerifyCountOfEstablishedConnectionsIs(allreadyEstablished + 1, "After second close");
+			}
 		}
 
 		[Test]
 		public void ConnectionPooling_OpenThenCloseTwoAtTheSameTime_TowConnectionsArePooled()
 		{
-			MakeDriver();
-
 			_driver.ClearPool(_connectionString);
 
 			var allreadyEstablished = GetEstablishedConnections();
 
-			var connection1 = MakeConnection();
-			var connection2 = MakeConnection();
-
-			//open first connection
-			connection1.Open();
-			VerifyCountOfEstablishedConnectionsIs(allreadyEstablished + 1, "After first open");
+			using (var connection1 = MakeConnection())
+			using (var connection2 = MakeConnection())
+			{
+				//open first connection
+				connection1.Open();
+				VerifyCountOfEstablishedConnectionsIs(allreadyEstablished + 1, "After first open");
 
-			//open second one
-			connection2.Open();
-			VerifyCountOfEstablishedConnectionsIs(allreadyEstablished + 2, "After second open");
+				//open second one
+				connection2.Open();
+				VerifyCountOfEstablishedConnectionsIs(allreadyEstablished + 2, "After second open");
 
-			//return connection1 to the pool
-			connection1.Close();
-			VerifyCountOfEstablishedConnectionsIs(allreadyEstablished + 2, "After first close");
+				//return connection1 to the pool
+				connection1.Close();
+				VerifyCountOfEstablishedConnectionsIs(allreadyEstablished + 2, "After first close");
 
-			//return connection2 to the pool
-			connection2.Close();
-			VerifyCountOfEstablishedConnectionsIs(allreadyEstablished + 2, "After second close");
+				//return connection2 to the pool
+				connection2.Close();
+				VerifyCountOfEstablishedConnectionsIs(allreadyEstablished + 2, "After second close");
+			}
 		}
 
 		[Test]
 		public void AdjustCommand_StringParametersWithinConditionalSelect_ThenParameterIsWrappedByAVarcharCastStatement()
 		{
-			MakeDriver();
-			var cmd = BuildSelectCaseCommand(SqlTypeFactory.GetString(0));
-
-			_driver.AdjustCommand(cmd);
+			using (var cmd = BuildSelectCaseCommand(SqlTypeFactory.GetString(255)))
+			{
+				_driver.AdjustCommand(cmd);
 
-			var expectedCommandTxt = "select (case when col = @p0 then cast(@p1 as VARCHAR(255)) else cast(@p2 as VARCHAR(255)) end) from table";
-			Assert.That(cmd.CommandText, Is.EqualTo(expectedCommandTxt));
+				var expectedCommandTxt =
+					"select (case when col = @p0 then cast(@p1 as VARCHAR(4000)) else cast(@p2 as VARCHAR(4000)) end) from table";
```

Comment:
It was also failing because still testing for `VARCHAR(255)` in resulting string, while the driver has been changed to use the default length instead of the supplied one for variable length string, due to the sql `like` issue otherwise.